### ISSUE TITLE
Unable to run intent discovery

### ISSUE THREAD

**Issue by SShah30-hue** [thuiar/TEXTOIR#34]
Hi, 

Many thanks for this repo. I ran intent detection and it runs smoothly. However, I am facing issues with intent discovery. The requirements were the first issue I encountered as some of the libraries were not getting installed. I then used python 3.7 and all installations were completed except deepspeed and triton. I manually installed deepspeed==0.3.16 but could'nt get triton installed. Regardless, I continued along the tutorial and ran run.py file but got below error. Please can you offer any advice? Thanks

Below is the syntax I used:
`python run.py --setting semi_supervised --dataset banking --known_cls_ratio 0.75 --train --save_results --results_file_name results_DeepAligned.csv --save_model --labeled_ratio 0.2`

![error](https://user-images.githubusercontent.com/61804269/174387695-7b5e6354-be2b-402a-9e53-034de1c5c3db.png)

**Comment by HanleiZhang** [comment]
It is because you failed to load the bert model. You should place the bert model in the correct path as in the open intent detection module :)

**Comment by SShah30-hue** [comment]
Hi Hanlei, thanks for your reply. I understand but it is already using the same path as open intent detection module which is `/home/sharing/disk1/pretrained_embedding/bert/uncased_L-12_H-768_A-12/`

**Comment by HanleiZhang** [comment]
I see. We have updated the open intent detection module so it can automatically download the needed pre-trained bert model. 
You can download the pre-trained bert model prepared in [ADB repository]( https://github.com/thuiar/Adaptive-Decision-Boundary) and put it in the correct path. ([Baidu Cloud Drive](https://pan.baidu.com/s/1k1zxK4xh0UyPhOU_-oPlow) with code: v8tk). By the way, the open intent discovery module will also be updated for more convenience within the next two months.

**Comment by SShah30-hue** [comment]
Thanks so much! I have a different question now - for test set how can I decode both prediction and ground truth labels into original labels and also show it's corresponding sentence and save that as pred file? Currently, it's storing y_pred.npy and y_true.npy but it's encoded and does not show corresponding sentence.

**Comment by SShah30-hue** [comment]
I have a different issue now - I am getting an error when setting the cluster_num_factor > 1 for intent discovery module. Below is the error:
![error](https://user-images.githubusercontent.com/61804269/176653281-06b4f871-36ec-4267-9210-a642c289da91.PNG)

Can you explain why this is happening?

**Comment by mrFocusXin** [comment]
Hello, we have updated the code in time, and the problem has been solved 

> I have a different issue now - I am getting an error when setting the cluster_num_factor > 1 for intent discovery module. Below is the error: ![error](https://user-images.githubusercontent.com/61804269/176653281-06b4f871-36ec-4267-9210-a642c289da91.PNG)
> 
> Can you please explain why this is happening?

Hello, we have updated the code in time, and the problem has been solved

**Comment by SShah30-hue** [comment]
Hi Hanlei,

Hope you are well.

Sorry it took me long to come back to you. I can see that the
open_intent_discovery module is still facing the issue of model loading.
Would it be possible for you to show exactly where I have to make changes
so that it automatically loads the model just like open_intent_detection
does?

Thank you in advance.

Kind regards,

Shariq
